### ISSUE TITLE
Clippy reports errors from macro even if they are marked with allow(clippy::all)

### ISSUE THREAD

**Issue by DrSloth** [rust-lang/rust-clippy#11132]
### Summary

This error occurred while writing code with `sqlx` in the macro expansion the code is marked with `#[allow(clippy::all)]` but the lints `arithmetic_side_effects` and `missing_panic_docs` still fire.

The second one might still make sense to keep because it CAN panic but in the sqlx sense that should never happen.

### Reproducer

I tried this code:
Any sqlx query with bind parameters should work

```rust
sqlx::query_as!(Foo, "SELECT boo, baz FROM Foos WHERE boo > $1 AND baz < $2", 10, 20)
```

I expected to see this happen:
No errors reported

Instead, this happened:
Clippy reports `arithmetic_side_effects` because it increments the bind parameter count. Also it reports `missing_panic_docs` because sqlx panics if the database schema is wrong. 

### Version

```text
rustc 1.70.0 (90c541806 2023-05-31)
binary: rustc
commit-hash: 90c541806f23a127002de5b4038be731ba1458ca
commit-date: 2023-05-31
host: x86_64-unknown-linux-gnu
release: 1.70.0
LLVM version: 16.0.2
```

**Comment by Centri3** [comment]
`clippy::all` only includes all that are warn/deny, so those lints aren't included. You may want to try to allow those specific lints instead.

**Comment by DrSloth** [comment]
Thanks, how can i allow this if it comes from a macro invocation the attribute on the macro invocation is ignored.

**Comment by c410-f3r** [comment]
https://github.com/rust-lang/rust-clippy/pull/10203, that is included in version 1.71, should fix this problem.

**Comment by Alexendoo** [comment]
closing per https://github.com/rust-lang/rust-clippy/issues/11132#issuecomment-1633399781, if this is still an issue though please re-open
